lib/Doctrine/ORM/Query/Expr/Between.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Query\Expr;
+
+final class Between
+{
+    public const NOT_BETWEEN = 'NOT BETWEEN';
+
+    public const BETWEEN = 'BETWEEN';
+
+    /** @var string */
+    private $operator;
+
+    /** @var int|string */
+    private $key;
+
+    /** @var int|string */
+    private $min;
+
+    /** @var int|string */
+    private $max;
+
+    /**
+     * @param string     $operator
+     * @param int|string $key
+     * @param int|string $min
+     * @param int|string $max
+     */
+    public function __construct($operator, $key, $min, $max)
```

Comment:
We should probably add assertions for these parameters. (to restrict their type according to this signature)

Reply:
What exactly do you mean by that? 

I can restrict the property "operator" with the type hinting string.